project-backend/Providers/JobProvider/JobProvider.cs

```diff
@@ -33,6 +34,25 @@ public JobDAO AddJob(string name, string description, int userId)
             return newJob;
         }
 
+        public JobDAO EditJob(int id, string newName, string newDescription, int userId)
+        {
+            JobDAO correspondingJob = _dbContext.Jobs.Find(id);
+
+            if (correspondingJob == null)
+                throw new ResourceNotFoundException("Job not found");
+
+            if (correspondingJob.UserId != userId)
+                throw new UnauthorizedException("User isn't the author of the job");
+
+            correspondingJob.Name = newName;
+            correspondingJob.Description = newDescription;
+
+            JobDAO updatedJob = _dbContext.Jobs.Update(correspondingJob).Entity;
+            _dbContext.SaveChanges();
+
+            return updatedJob;
```

Comment:
```suggestion
            _dbContext.SaveChanges();

            return correspondingJob;
```
No need to call Update, as the entity is managed by the framework

Reply:
Nice, didn't know that.